integration-tests/trait-dyn-cross-contract-calls/lib.rs

```diff
@@ -0,0 +1,106 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// We can't run E2E test without the contract. So add `Dummy` contract.
```

Comment:
Is this the same issue as https://github.com/paritytech/ink/issues/1672

Reply:
No, it is another reason. During running of the `cargo test --features e2e-tests` ink! tries to build the current file as a contract via `cargo-contract`. But if it is not a contract, it will fail because it doesn't have `___ink_generate_metadata` function=)

Reply:
That seems like a bug, we shouldn't need the metadata anymore for `ink_e2e`

Reply:
Anyway we should attempt to make this more ergonomic for structuring `e2e` tests in this way

Reply:
This prompted me to open https://github.com/paritytech/ink/pull/1683 which means it doesn't need to generate the metadata anymore. 

Not sure it will make this structure work without the dummy contract, but at least it is an immprovement.

Reply:
See #1691 for a possible solution